lib/service.js

```diff
@@ -3091,7 +3104,99 @@
             });
             
             return req;
-        }
+        },
+        
+        doesFileExist: async function(filename) {
+            try {
+                var that = this;
+
+                // 1. Fetch files list
+                var response = await this.get("",  {__conf: filename}, null, true);
+
+                // 2. Filter the files
+                var files = response 
+                                && response.body 
+                                && response.body.entry 
+                                && response.body.entry.filter(f => f.name === filename);
+                
+                // 3. Check if the file exists
+                if(files && files.length == 0) {
+                    return { code:"FILE_NOT_FOUND", message: `File with the name '${filename}' does not exist.`};
+                }
+
+                // 4. Create a local instance
+                var configurationFile = new root.ConfigurationFile(that.service, filename);
+
+                // 5. Load the file content
+                var fetchedFile = await configurationFile.fetchAsync();
+
+                return {file: fetchedFile};
+            }
+            catch(error) {
+                return { errorCode:"SYSTEM_ERROR", errorMessage: error};
+            }
+        },
+
+        doesStanzaExist: async function(file, stanza) {
+            try {
+                
+                // 1. check if the stanza exists
+                var fetchedStanza = file.item(stanza);
+                
+                if(!fetchedStanza) {
+                    return { code:"STANZA_NOT_FOUND", message: `Stanza with the name '${stanza}' does not exist.`};
+                }
+                else {
+                    return {stanza: fetchedStanza};
+                }
+            }
+            catch(error) {
+                return { errorCode:"SYSTEM_ERROR", errorMessage: error};
+            }
+        },
+
+        createAsync: async function (filename, stanza, keyValueMap, callback) {
+            callback = callback || function() {};
+            var that = this;
+
+            // 1. Check if the file exists
+            var doesFileExistResponse = await this.doesFileExist(filename);
+            var configFile = doesFileExistResponse.file;
+
+            // 2. If the file doesn't exist, create a new file
+            if(!configFile) {
```

Comment:
nit: I think `configFile === undefined` might be slightly preferably here - same for `if(!configStanza)` below on line 3183

Reply:
Done.